src/app/test-mode/test-view/test-view.component.ts

```diff
@@ -0,0 +1,101 @@
+import {Component, OnInit} from '@angular/core';
+import {ActivatedRoute, Router} from "@angular/router";
+import {CollectionService} from "../shared/collection.service";
+import {DeckDto} from "../../collections/shared/dtos/deck/deck.dto";
+import {CardDto} from "../../collections/shared/dtos/card/card.dto";
+import {SummaryService} from "../shared/summary.service";
+
+@Component({
+  selector: 'app-test-view',
+  templateUrl: './test-view.component.html',
+  styleUrls: ['./test-view.component.css']
+})
+export class TestViewComponent implements OnInit {
+
+  deckId : number | undefined;
+  deck: DeckDto | undefined;
+  cards: CardDto[] | undefined;
+  currentCardIndex: number = 0;
+
+  correct: number = 0;
+  wrong: number = 0;
+  skipped: number = 0;
+  answer: string | undefined;
+
+  correctCards: Array<CardDto> = [];
+  wrongCards: Array<CardDto> = [];
+  skippedCards: Array<CardDto> = [];
+
+  checked: boolean = false;
+  answeredCorrectly: boolean | undefined;
+
+  constructor(private route: ActivatedRoute, private service: CollectionService,
+              private router: Router, private summaryService: SummaryService) { }
+
+  ngOnInit(): void {
+    // @ts-ignore
```

Comment:
i used it too but maybe later we can come up with something better than // @ts-ignore